### PR TITLE
fix(vcs): nsfw property v13 backport

### PR DESCRIPTION
**Pull request by CodeGoat-dev**
**Please describe the changes this PR makes and why it should be merged:**

Backports #8132 to v13.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: src/structures/BaseGuildTextChannel.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ class BaseGuildTextChannel extends GuildChannel {
      * If the guild considers this channel NSFW
      * @type {boolean}
      */
-    this.nsfw = Boolean(data.nsfw);
```
Comment (KinectTheUnknown) [inline src/structures/BaseGuildTextChannel.js:33]: I wouldn't remove the Boolean call since ur using it in VoiceChannel

**Review by KinectTheUnknown — CHANGES_REQUESTED** [review]

File: src/structures/BaseGuildTextChannel.js (hunk 1 above)
Comment (CodeGoat-dev) [reply src/structures/BaseGuildTextChannel.js]: I've resolved that.

File: src/structures/VoiceChannel.js [hunk 2]
```diff
@@ -59,6 +65,18 @@ class VoiceChannel extends BaseGuildVoiceChannel {
...
       this.rateLimitPerUser = data.rate_limit_per_user;
     }
+
+    if ('nsfw' in data) {
+      /**
+       * The rate limit per user (slowmode) for this channel in seconds
+       * @type {number}
+       */
+      this.rateLimitPerUser = data.rate_limit_per_user;
+    }
```
Comment (Jiralite) [inline src/structures/VoiceChannel.js:75]: What's this doing here?

**Review by Jiralite — CHANGES_REQUESTED** [review]
This seems to be missing typings, no?

File: src/structures/BaseGuildTextChannel.js [hunk 3]
```diff
@@ -47,7 +47,11 @@ class BaseGuildTextChannel extends GuildChannel {
     }
 
     if ('nsfw' in data) {
-      this.nsfw = Boolean(data.nsfw);
+      /**
+       * If the guild considers this channel NSFW
+       * @type {boolean}
+       */
```
Comment (Jiralite) [inline src/structures/BaseGuildTextChannel.js:53]: The JSDoc for this property is now duplicated. It is a couple of lines above this.

File: src/structures/VoiceChannel.js (hunk 2 above)
Comment (Jiralite) [reply src/structures/VoiceChannel.js:75]: This is not resolved @CodeGoat-dev.

**Review by Jiralite — CHANGES_REQUESTED** [review]
Typings are still missing.

Look at the pull request you are trying to backport for pointers.

**Pull request closed without merging**